lib/std/fs/path.zig

```diff
@@ -1319,3 +1319,40 @@ test "extension" {
     try testExtension("/foo/bar/bam/a.b.c", ".c");
     try testExtension("/foo/bar/bam/a.b.c/", ".c");
 }
+
+/// Returns the stem of the given file name (if any).
+/// Examples:
+/// - "hello/world/lib.tar.gz" ⇒ "lib.tar"
+/// - "hello/world/lib.tar"    ⇒ "lib"
+/// - "hello/world/lib"        ⇒ "lib"
+pub fn stem(path: []const u8) []const u8 {
+    const filename = basename(path);
+    const index = mem.lastIndexOfScalar(u8, filename, '.') orelse return filename[0..];
+    if (index == 0) return path;
+    return filename[0..index];
+}
```

Comment:
You could also inline the calls to `self.basename(self.extension)` here. 4 LOC vs 2 is not a huge win though.

Reply:
See above. If you're saying this function could be `retun basename(extension(path));` then I don't think that's right because the tests don't pass.